File: Code/new_fitervari_backend/src/main/java/at/htl/control/NewsLetterRepository.java
package at.htl.control;

import at.htl.model.NewsLetter;
import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class NewsLetterRepository implements PanacheRepository<NewsLetter> {

    public NewsLetter save(NewsLetter entity) {
        return getEntityManager().merge(entity);
    }
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/repository/NewsLetterRepository.kt
package at.htl_leonding.repository

import at.htl_leonding.model.NewsLetter
import io.quarkus.hibernate.orm.panache.PanacheRepository
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject
import javax.persistence.EntityManager

@ApplicationScoped
class NewsLetterRepository: PanacheRepository<NewsLetter> {
    @Inject
    lateinit var em: EntityManager

    fun save(newsLetter: NewsLetter): NewsLetter {
        return em.merge(newsLetter)
    }
}